File: src/main/java/com/cfgglobal/test/dao/PermissionDao.java
package com.cfgglobal.test.dao;

import com.cfgglobal.test.dao.base.BaseDao;
import com.cfgglobal.test.domain.Permission;
import io.vavr.collection.List;
import io.vavr.control.Option;
import org.springframework.stereotype.Repository;

@Repository
public interface PermissionDao extends BaseDao<Permission, Long> {

    Option<Permission> findByHttpMethod(String httpMethod);

    List<Permission> findByEntity(String entity);
}

File: src/main/kotlin/com/cfgglobal/test/dao/PermissionDao.kt
package com.cfgglobal.test.dao

import com.cfgglobal.test.dao.base.BaseDao

import com.cfgglobal.test.domain.Permission
import org.springframework.stereotype.Repository

@Repository
interface PermissionDao : BaseDao<Permission, Long> {

    fun findByHttpMethod(httpMethod: String): Permission

    fun findByEntity(entity: String): List<Permission>
}